internal/backend/rclone/backend.go

```diff
@@ -85,6 +85,9 @@ func run(command string, args ...string) (*StdioConn, *sync.WaitGroup, func() er
 		err = errW
 	}
 	if err != nil {
+		if errors.IsErrDot(err) {
+			return nil, nil, nil, errors.Errorf("run: cannot implicitly run relative executable %v found in current directory, use -o rclone.program with ./ to override", cmd.Path)
```

Comment:
it would be possible to explicitly list the necessary option in the error message:
`errors.Errorf("cannot implicitly run executable %v found in current directory, use '-o rclone.program=./%v' to override", cmd.Path, cmd.Path)` (the "run:" prefix doesn't seem necessary as the source of the error is clear)

Reply:
Yeah, I actually tried this, but found that it just made things overly complicated when people use quotes somewhere in their `rclone.program` or `sftp.command`.

**TL;DR:** Unless we can print the actual syntax that the user entered in their shell for the extended option (which we can't since by the time we get information about the extended option, the original syntax has been lost), it will be hard to give them a good or even working example of what they should run instead.

Taking SFTP as a (more obvious than rclone) example and a user running `restic -o sftp.command="ssh -o 'VerifyHostKeyDNS ask'" snapshots` (or whatever option to `ssh`, this is just a dummy example):
- We'd need to make the error message surround the `./%v` with quotes, so the code would be like: `errors.Errorf("startClient: cannot implicitly run relative executable %v found in current directory, use -o sftp.command='./%v' to override", cmd.Path, cmd.String())` or with `""` instead of `''`.
- Regardless of surrounding quote type, we don't get quoting information out of `Cmd.String()`, so when we print the error it becomes `.. use -o sftp.command='./ssh -o VerifyHostKeyDNS ask' to override`, losing the quotes around `VerifyHostKeyDNS ask`.

We'd have to do quite some deduction to print an extended option that is the same as the user originally entered with with `./` before it. Unless you know a way to do this, I'm open to ideas.

That said, my rationale was that `use -o sftp.command with ./ to override` would be clear enough. But perhaps `use -o sftp.command prefixed with ./ to override` or even `prefix the value of -o sftp.command with ./ to override` is even clearer. Some people might not know what `./` is and hence not know how to use it/where to place it.

Reply:
Regarding the prefix, errors seems wrapped throughout restic source code so I figured it would be good to do that here too. But if you want me to remove it I'm fine with that!

Reply:
Hmm, is the quoting issue actually relevant when using the `rclone` backend? Unless someone uses quotes in the filename, things won't break... Or as an alternative what about `use '-o rclone.program=./<cmd>' to override` instead? That could make it more obvious what the syntax to use the option is.

Regarding sftp, after looking at the code, I'm a bit unsure what command I'd have to use. The auto-generated command is `ssh $host [-p $port] [-l $user] -s sftp`. So, I don't think a simple `-o sftp.command='./ssh'` can work at all?

Reply:
> Hmm, is the quoting issue actually relevant when using the `rclone` backend?

It may not be a problem in the case of rclone, but I just never saw the point in risking it. I also don't see the point in making the rclone and sftp error messages different (except the option name), hence me trying to keep them under the same reasoning.

Rclone does have for example the `--config <path>` option which when having a space in ones username means one'll have to quote that path or escape the space in it (I'd say most people would quote it). I suppose people can put that in `rclone.args` though, but there's no corresponding `sftp.args`.

> Or as an alternative what about use `'-o rclone.program=./<cmd>' to override` instead? That could make it more obvious what the syntax to use the option is.

I'm not sure I agree :) First off, this entire discussion is about catering to those users that don't already, after just reading the slightly better error message and seeing the words "use" and "./" along with the option name, understands what they need to do. For those users, I dare say that some will think that they should now wrap the entire option in quotes (`'-o rclone.program=./<cmd>'` as opposed to `-o rclone.program=./<cmd>`). So the quotes should probably not be there. Second, if a user doesn't understand how to put `./` in the value of these options, I'm pretty sure they'd also start wondering "what \<cmd>/command should I use instead?". And speaking about quotes, `use -o rclone.program=./<cmd> to override` might, sadly, make some users think that they can write an actual command with spaces in place of `<cmd>` without quoting it, since there are no quotes in the suggestion.

This is why I figured "keep it simple" and that instead of trying to work around every possible misinterpretation of the suggestion in the error message, just saying "add a ./ before the value" would be clear enough.

Either way, let's do what you want here, I think there are more important things to discuss than this. What about `use -o rclone.program=./<cmd> to override` (that is, without the quotes)? And the equivalent for sftp of course.

Reply:
> Rclone does have for example the `--config <path>` option which when having a space in ones username means one'll have to quote that path or escape the space in it (I'd say most people would quote it)

Looks like I should look into the code first. I've expected `rclone.program` to only contain the binary name, but not also possibly some flags from rclone.

> This is why I figured "keep it simple" and that instead of trying to work around every possible misinterpretation of the suggestion in the error message, just saying "add a ./ before the value" would be clear enough.

Let's keep it at the current wording then and just see what happens. There seem to be just too many corner cases to simply provide a proper error message. ("%q" would probably be enough to handle quoting for unix-like systems, but I have my doubts about windows)

Reply:
> I've expected rclone.program to only contain the binary name, but not also possibly some flags from rclone.

But you're pretty much right in your expectation there, I think the purpose of `rclone.args` *is* to put stuff like `--config` in there instead of in `rclone.program`. However, now that you mentioned it, I realize that users might put it in `rclone.program` anyway, in which case we'd be back to it being hard to suggest them the correct option value.

> "%q" would probably be enough to handle quoting for unix-like systems, but I have my doubts about windows

Unfortunately even `%q` wouldn't yield the right message, e.g. `-o sftp.command="ssh -o 'VerifyHostKeyDNS ask'"` would be printed as `-o sftp.command="ssh -o VerifyHostKeyDNS ask"` as it just quotes the entire value.

> Let's keep it at the current wording then and just see what happens.

I changed it to a variant of your suggestion, it's now `use -o sftp.command=./<command> to override` and `use -o rclone.program=./<program> to override` :) I realized that my suggestion of `prefix the value of ..` is bad because users might not even have specified the extended option in the first place, so referring to that isn't very user friendly.